style_guide.md

```diff
@@ -0,0 +1,286 @@
+          when: [rules.isNotProxy],
+        },
+        {
+          question: question(isProxy=true),
+          when: [rules.isProxy],
+        },
+      ],
+    }
+    ```
+- Variants using combination of `isProxy` and `isEmployed`:
+
+    ```
+    {
+      type: 'Question',
+      id: 'business-name',
+      question_variants: [
+        {
+          question: question(isProxy=false, isEmployed=true),
+          when: [rules.isNotProxy, rules.isEmployed],
+        },
+        {
+          question: question(isProxy=true, isEmployed=true),
+          when: [rules.isProxy, rules.isEmployed],
+        },
+        {
+          question: question(isProxy=false, isEmployed=false),
+          when: [rules.isNotProxy, rules.isNotEmployed],
+        },
+        {
+          question: question(isProxy=true, isEmployed=false),
+          when: [rules.isProxy, rules.isNotEmployed],
+        },
+      ],
+    }
+    ```
+- In the question method, when we need to resolve title, we call a new questionTitle method, passing `isProxy`:
+    ```
+    local question(isProxy) = {
+      title: questionTitle(isProxy)
+      id: 'confirm-date-of-birth',
+      title: title,
+      type: 'General',
+      answers: [
+      {
+        id: 'confirm-date-of-birth-answer',
+        mandatory: true,
+        options: [
+          {
+            label: 'Yes',
+            value: 'Yes',
+           },
+          {
+            label: 'No',
+            value: 'No',
+          },
+        ],
+        type: 'Radio',
+        },
+      ],
+    };
+    ```
+- In the `questionTitle` method we would return either of the two strings, based on the `isProxy` variable:
+    ```
+    local questionTitle(isProxy) = (
+      if isProxy == true then {
+      text: 'What was <em>{person_name_possessive}</em> age on their last birthday?',
+      placeholders: [
+        placeholders.personNamePossessive,
+      ]} 
+      else 'What was your age on your last birthday?';
+    );
+    ```
+- Example of `questionTitle` with multiple variables:
+    ```
+    local questionTitle(isProxy, isEmployed) = (
+      if isProxy == true && isEmployed == true then {
+      text: 'What is the main activity of <em>{person_name_possessive}</em> organisation, business or freelance work?',
+      placeholders: [
+        placeholders.personNamePossessive,
+      ]}
+      else if isProxy == false && isEmployed == true then {
+      text: 'What is the main activity of <em>{person_name_possessive}</em> organisation, business or freelance work?',
+      placeholders: [
+        placeholders.personNamePossessive,
+      ]}
+      else if isProxy == true && isEmployed == false then 'What was the main activity of your organisation, business or freelance work?'
+      else {
+      text: 'What was the main activity of <em>{person_name_possessive}</em> organisation, business or freelance work?',
+      placeholders: [
+        placeholders.personNamePossessive,
+      ]};
+    );  
```

Comment:
This isn't very clear. We need to think of a way to structure this to make it more readable.

Reply:
Made it clear as per your suggestion, thanks!